cloud/cloudstack/cs_instance.py

```diff
@@ -157,6 +157,11 @@
     required: false
     default: []
     aliases: [ 'affinity_group' ]
+  start_vm:
+    description:
+      - Whether to start the instance after first deployment.
+    required: false
+    default: true
```

Comment:
Good question, not really:

`startvm` is provided by the API on the command `deployVirtualMachines` only, to deploy a VM but have the possibility to specify whether to start the VM after deployment. The default is `startvm=true`. I implemented this, so you can make sure a VM exists, but do not boot it, if it need to be deployed.

This means if you say by ansible `state=present` and `start_vm=true`, a stopped deployed (ergo present) VM will not be started. This sounds a bit confusing, but it is the way the API works. I tried to make it clear in the description. I know the `start_vm` name is a bit misleading but it is identical to the API, that is why I took it. I agree a name like `start_after_deploy` or `boot_after_deploy` would be much clearer. Any thoughts?

ansible `state=present` does not `start` a VM, it deploys a VM if not available and cloudstack starts the VM after deployment (because startvm=true by default by the API). However, if the VM exists (whether it is running or stopped) `state=present` does return `changed=false`

ansible `state=started` starts a stopped VM only, it does not deployment. This means, the VM must exists.